server/middleware/session/setSession.js

```diff
@@ -75,8 +77,26 @@ const setSession = async (req, res, next) => {
   } catch (e) {
     // console.error('Jwt Verify failed', e)
 
-    if (e.name === 'TokenExpiredError') {
-      res.clearCookie('idToken')
+    if (e instanceof TokenExpiredError) {
+      // Don't redirect user to an /api/ path after signing through,
+      // the user would be treated with a nice wall of json.
+      // This should only happen if the access token expires after page load but before ajax is exectued on a page.
+      const redirectUrl = req.url.startsWith('/api/') ? '/home/' : req.url
+
+      const qs = queryString.stringify({ redirect: redirectUrl })
```

Comment:
Unsure if queryString library is the right choice. I saw it being used elsewhere.

encodeURIComponent would work also.

Reply:
yes please use encodeURIComponent